x/dex/genesis.go

```diff
@@ -11,41 +11,72 @@ import (
 func InitGenesis(ctx sdk.Context, k keeper.Keeper, genState types.GenesisState) {
 	k.CreateModuleAccount(ctx)
 
+	lastEpoch := uint64(0)
+
 	// Set all the longBook
-	for _, elem := range genState.LongBookList {
-		k.SetLongBook(ctx, "genesis", elem)
-	}
+	for _, contractState := range genState.ContractState {
+		for _, elem := range contractState.LongBookList {
+			k.SetLongBook(ctx, contractState.ContractInfo.ContractAddr, elem)
+		}
 
-	// Set all the shortBook
-	for _, elem := range genState.ShortBookList {
-		k.SetShortBook(ctx, "genesis", elem)
-	}
+		for _, elem := range contractState.ShortBookList {
+			k.SetShortBook(ctx, contractState.ContractInfo.ContractAddr, elem)
+		}
 
-	for _, elem := range genState.TriggeredOrdersList {
-		// not sure if it's guaranteed that the Order has the correct Price/Asset/Contract details...
-		k.SetTriggeredOrder(ctx, "genesis", elem, elem.PriceDenom, elem.AssetDenom)
-	}
+		for _, elem := range contractState.TriggeredOrdersList {
+			// not sure if it's guaranteed that the Order has the correct Price/Asset/Contract details...
+			k.SetTriggeredOrder(ctx, contractState.ContractInfo.ContractAddr, elem, elem.PriceDenom, elem.AssetDenom)
+		}
+
+		// Set initial tick size for each pair
+		// tick size is the minimum unit that can be traded for certain pair
+		for _, elem := range contractState.PairList {
+			// TODO:(kartik) Is this needed since tick size already part of pair?
+			// This might be necessary because tick size store is keyed by pricedenom/assetdenom not full struct
+			k.SetDefaultTickSizeForPair(ctx, elem, *elem.Ticksize)
```

Comment:
we need to set tick size, but `SetDefaultTickSizeForPair` is deprecated. Let's use `SetTickSizeForPair`

Reply:
Updated on latest commit

Reply:
can you rebase the quantity tick size PR and update this as well

Reply:
Rebased off the tick size PR (and set the github PR base to that PR as well)